amazon-kinesis-client/src/test/java/software/amazon/kinesis/utils/TestConsumer.java

```diff
@@ -0,0 +1,202 @@
+package software.amazon.kinesis.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.RandomStringUtils;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+import software.amazon.awssdk.services.kinesis.model.PutRecordRequest;
+import software.amazon.kinesis.checkpoint.CheckpointConfig;
+import software.amazon.kinesis.common.ConfigsBuilder;
+import software.amazon.kinesis.common.InitialPositionInStreamExtended;
+import software.amazon.kinesis.common.KinesisClientUtil;
+import software.amazon.kinesis.config.KCLAppConfig;
+import software.amazon.kinesis.coordinator.CoordinatorConfig;
+import software.amazon.kinesis.coordinator.Scheduler;
+import software.amazon.kinesis.leases.LeaseManagementConfig;
+import software.amazon.kinesis.lifecycle.LifecycleConfig;
+import software.amazon.kinesis.metrics.MetricsConfig;
+import software.amazon.kinesis.processor.ProcessorConfig;
+import software.amazon.kinesis.retrieval.RetrievalConfig;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+@Slf4j
+public class TestConsumer {
```

Comment:
It looks like this framework does not teardown/delete the Kinesis stream or DDB?

Please ensure the following:
1. Test resources are deleted on teardown -- both on test success, and test failure (e.g., thrown Exception that fails the whole process).
    * We don't want inert test resources to continue stockpiling.
    * We don't want test resources to be reused (e.g., DDB) which may contaminate a test-run leading to either a false success or false failure.
3. Resources created per test are unique. Your [comment](https://github.com/awslabs/amazon-kinesis-client/pull/1136#discussion_r1228550016) highlighted a problem w/ the non-uniqueness.
    * Uniqueness helps avoid test contamination.
    * Uniqueness allows parallelization of multiple tests. (Having one name per test partially addresses this, but what if we need to execute multiple tests of the same "name"?)

Reply:
1. Implemented resource teardown on success or fail of test case. 
2. Resources created (stream and DDB) both have name getStreamName(). As long as these are unique per test case, there will be no issues with overlap. Not really sure that I understand your point about "executing multiple tests of the same name"